engine/src/main/java/org/terasology/rendering/nui/layers/ingame/metrics/DebugOverlay.java

```diff
@@ -167,8 +163,10 @@ public Boolean get() {
 
     @Override
     public void update(float delta) {
-        if (metricsLabel != null) {
-            metricsLabel.setText(metricsModes.get(currentMode).getMetrics());
+        if (debugRendering.isMetricModesAvailable()) {
+            metricsLabel.setText(debugRendering.getCurrentMode().getMetrics());
+        } else {
+            metricsLabel.setText("No metric modes are available!");
```

Comment:
Instead of explicitly checking whether the list is empty of not, you could add a default dummy mode and guarantee that `getCurrentMode` is never `null`.